x-kiosk-imx8/Dockerfile

```diff
@@ -0,0 +1,225 @@
+COPY libg2d1_6.4.3.p1.4-r0_arm64.deb /
+COPY libdrm2_2.4.102.imx-r0_arm64.deb /
+COPY imx-gpu-viv_6.4.3.p1.4-aarch64-r0_arm64.deb /
+
+RUN \
+    dpkg -i libgal-imx_6.4.3.p1.4-aarch64-r0_arm64.deb && \
+    dpkg -i libg2d1_6.4.3.p1.4-r0_arm64.deb && \
+    dpkg -i libdrm2_2.4.102.imx-r0_arm64.deb  && \
+    dpkg -i imx-gpu-viv_6.4.3.p1.4-aarch64-r0_arm64.deb
+
+RUN \
+    mkdir -p /usr/lib/chromium/ && \
+    mkdir -p /share/icons/hicolor/16x16/apps/ && \
+    mkdir -p /share/icons/hicolor/22x22/apps/ && \
+    mkdir -p /share/icons/hicolor/24x24/apps/ && \
+    mkdir -p /share/icons/hicolor/32x32/apps/ && \
+    mkdir -p /share/icons/hicolor/48x48/apps/ && \
+    mkdir -p /share/icons/hicolor/64x64/apps/ && \
+    mkdir -p /share/icons/hicolor/128x128/apps/ && \
+    mkdir -p /share/icons/hicolor/256x256/apps/ && \
+    mkdir -p /usr/lib/chromium/locales
+
+COPY --from=build /chromium/chromium-96.0.4664.110/chrome/app/theme/default_100_percent/chromium/product_logo_16.png /share/icons/hicolor/16x16/apps/chromium.png
+COPY --from=build /chromium/chromium-96.0.4664.110/chrome/app/theme/default_100_percent/chromium/product_logo_name_22.png /share/icons/hicolor/22x22/apps/chromium.png
+COPY --from=build /chromium/chromium-96.0.4664.110/chrome/app/theme/chromium/product_logo_24.png /share/icons/hicolor/24x24/apps/chromium.png
+COPY --from=build /chromium/chromium-96.0.4664.110/chrome/app/theme/default_100_percent/chromium/product_logo_32.png /share/icons/hicolor/32x32/apps/chromium.png
+COPY --from=build /chromium/chromium-96.0.4664.110/chrome/app/theme/chromium/product_logo_48.png /share/icons/hicolor/48x48/apps/chromium.png
+COPY --from=build /chromium/chromium-96.0.4664.110/chrome/app/theme/chromium/product_logo_64.png /share/icons/hicolor/64x64/apps/chromium.png
+COPY --from=build /chromium/chromium-96.0.4664.110/chrome/app/theme/chromium/product_logo_128.png /share/icons/hicolor/128x128/apps/chromium.png
+COPY --from=build /chromium/chromium-96.0.4664.110/chrome/app/theme/chromium/product_logo_256.png /share/icons/hicolor/256x256/apps/chromium.png
+COPY --from=build /chromium/chromium-96.0.4664.110/out/Default/*.pak /usr/lib/chromium/
+COPY --from=build /chromium/chromium-96.0.4664.110/out/Default/resources.pak  /usr/lib/chromium/resources.pak 
+COPY --from=build /chromium/chromium-96.0.4664.110/out/Default/locales/*.pak  /usr/lib/chromium/locales/
+COPY --from=build /chromium/chromium-96.0.4664.110/out/Default/chrome_crashpad_handler  /usr/lib/chromium/
+COPY --from=build /chromium/chromium-96.0.4664.110/out/Default/chrome /usr/lib/chromium/
+COPY --from=build /chromium/chromium-96.0.4664.110/out/Default/*.bin /usr/lib/chromium/
+COPY --from=build /chromium/chromium-96.0.4664.110/out/Default/icudtl.dat /usr/lib/chromium/
+COPY --from=build /chromium/chromium-96.0.4664.110/out/Default/chrome-wrapper /usr/lib/chromium/
+
+RUN ln -s /usr/lib/chromium/chrome-wrapper /usr/bin/chrome
+
+USER ${FIO_USERNAME}
+WORKDIR ${FIO_HOME}
```

Comment:
```suggestion
USER ${FIO_USERNAME}
WORKDIR ${FIO_HOME}

COPY libgal-imx_6.4.3.p1.4-aarch64-r0_arm64.deb /
COPY libg2d1_6.4.3.p1.4-r0_arm64.deb /
COPY libdrm2_2.4.102.imx-r0_arm64.deb /
COPY imx-gpu-viv_6.4.3.p1.4-aarch64-r0_arm64.deb /

RUN \
    groupadd -g 61 render && \
    if [ ! "${FIO_USERID}" = "0" ]; then \
        groupadd -g ${FIO_USERID} ${FIO_USERNAME}; \
        useradd -d ${FIO_HOME} -s /bin/bash -m ${FIO_USERNAME} -u ${FIO_USERID} -g ${FIO_USERID} -G video,render; \
    fi && \
    apt-get update && \
    apt-get install -y vim curl wget libglib2.0-dev libnss3 libatk1.0-dev \
                       libatk-bridge2.0-0 libjpeg-turbo8 libdrm-dev \
                       libxkbcommon0 libasound2 libflac8 libxslt1.1 \
                       fontconfig-config libwayland-cursor0 libwayland-client0 \
                       libwayland-egl1 libxdamage1 libxfixes3 strace && \
    apt-get install -y --no-install-recommends \
        libwayland-client0 \
        libwayland-server0 && \
    apt-get clean && apt-get autoremove && rm -rf /var/lib/apt/lists/*  && \
    dpkg -i libgal-imx_6.4.3.p1.4-aarch64-r0_arm64.deb && \
    dpkg -i libg2d1_6.4.3.p1.4-r0_arm64.deb && \
    dpkg -i libdrm2_2.4.102.imx-r0_arm64.deb  && \
    dpkg -i imx-gpu-viv_6.4.3.p1.4-aarch64-r0_arm64.deb && \
    mkdir -p /usr/lib/chromium/ && \
    mkdir -p /share/icons/hicolor/16x16/apps/ && \
    mkdir -p /share/icons/hicolor/22x22/apps/ && \
    mkdir -p /share/icons/hicolor/24x24/apps/ && \
    mkdir -p /share/icons/hicolor/32x32/apps/ && \
    mkdir -p /share/icons/hicolor/48x48/apps/ && \
    mkdir -p /share/icons/hicolor/64x64/apps/ && \
    mkdir -p /share/icons/hicolor/128x128/apps/ && \
    mkdir -p /share/icons/hicolor/256x256/apps/ && \
    mkdir -p /usr/lib/chromium/locales

COPY --from=build /chromium/chromium-96.0.4664.110/chrome/app/theme/default_100_percent/chromium/product_logo_16.png /share/icons/hicolor/16x16/apps/chromium.png
COPY --from=build /chromium/chromium-96.0.4664.110/chrome/app/theme/default_100_percent/chromium/product_logo_name_22.png /share/icons/hicolor/22x22/apps/chromium.png
COPY --from=build /chromium/chromium-96.0.4664.110/chrome/app/theme/chromium/product_logo_24.png /share/icons/hicolor/24x24/apps/chromium.png
COPY --from=build /chromium/chromium-96.0.4664.110/chrome/app/theme/default_100_percent/chromium/product_logo_32.png /share/icons/hicolor/32x32/apps/chromium.png
COPY --from=build /chromium/chromium-96.0.4664.110/chrome/app/theme/chromium/product_logo_48.png /share/icons/hicolor/48x48/apps/chromium.png
COPY --from=build /chromium/chromium-96.0.4664.110/chrome/app/theme/chromium/product_logo_64.png /share/icons/hicolor/64x64/apps/chromium.png
COPY --from=build /chromium/chromium-96.0.4664.110/chrome/app/theme/chromium/product_logo_128.png /share/icons/hicolor/128x128/apps/chromium.png
COPY --from=build /chromium/chromium-96.0.4664.110/chrome/app/theme/chromium/product_logo_256.png /share/icons/hicolor/256x256/apps/chromium.png
COPY --from=build /chromium/chromium-96.0.4664.110/out/Default/*.pak /usr/lib/chromium/
COPY --from=build /chromium/chromium-96.0.4664.110/out/Default/resources.pak  /usr/lib/chromium/resources.pak 
COPY --from=build /chromium/chromium-96.0.4664.110/out/Default/locales/*.pak  /usr/lib/chromium/locales/
COPY --from=build /chromium/chromium-96.0.4664.110/out/Default/chrome_crashpad_handler  /usr/lib/chromium/
COPY --from=build /chromium/chromium-96.0.4664.110/out/Default/chrome /usr/lib/chromium/
COPY --from=build /chromium/chromium-96.0.4664.110/out/Default/*.bin /usr/lib/chromium/
COPY --from=build /chromium/chromium-96.0.4664.110/out/Default/icudtl.dat /usr/lib/chromium/
COPY --from=build /chromium/chromium-96.0.4664.110/out/Default/chrome-wrapper /usr/lib/chromium/

RUN ln -s /usr/lib/chromium/chrome-wrapper /usr/bin/chrome
```

Reply:
This comment may not work, but that's the idea i'm pushing on.

Reply:
That shouldn't work because Weston user doesn't have root rights. You need to be root to do apt-get.

Reply:
@doanac I think I understood your point, the only difference is that I really prefer to leave group to add separated of installing commands. This is because we use to use groupadd block on other containers that use Weston. So it is easy to identify if we have to change it in the future:

```
RUN if [ ! "${FIO_USERID}" = "0" ]; then \
        groupadd -g ${FIO_USERID} ${FIO_USERNAME}; \
        useradd -d ${FIO_HOME} -s /bin/bash -m ${FIO_USERNAME} -u ${FIO_USERID} -g ${FIO_USERID} -G video,render; \
    fi
```

Take a look how it is now. I think it is much better.